### PR TITLE
usm: Enable event monitoring if USM is enabled and loaded

### PR DESCRIPTION
**Pull request by guyarb**
### What does this PR do?

Fixing a possible panic in the code base and a bug in which the USM event monitor can run without USM.

The fix is to enable USM's event monitoring if the appropriate configuration is set (like today) but also to check USM is enabled (to prevent a case in which USM is disabled, event monitoring is enabled, and we execute redundant code). Also, we initialize USM event monitor only if the network tracer successfully started

### Motivation

If USM was enabled with event monitoring, but NPM failed to start (for instance, conntracker failure) the network tracer component didn't start, but the event monitoring of USM would start, and once we got an event of a process execution or termination, we would try to update our telemetry object which was never initialized, causing a panic as some fields are pointers (nil pointers).

The bug revealed both a possible panic in the code, but also the fact that event monitoring of USM can run even if USM failed to start.

### CODE REVIEW

File: cmd/system-probe/modules/eventmonitor.go [hunk 1]
```diff
@@ -70,6 +70,18 @@ func createEventMonitorModule(_ *sysconfigtypes.Config, wmeta optional.Option[wo
 		}
 		evm.RegisterEventConsumer(network)
 		log.Info("event monitoring network consumer initialized")
+
+		netconfig := netconfig.New()
+		if netconfig.EnableUSMEventStream && netconfig.ServiceMonitoringEnabled {
```
Comment (hmahmood) [inline cmd/system-probe/modules/eventmonitor.go:75]: Isn't this already checked here? https://github.com/datadog/datadog-agent/blob/main/cmd/system-probe/modules/eventmonitor_linux.go#L31

File: pkg/network/tracer/tracer.go [hunk 2]
```diff
@@ -182,6 +182,11 @@ func newTracer(cfg *config.Config, telemetryComponent telemetryComponent.Compone
 	tr.reverseDNS = newReverseDNS(cfg, telemetryComponent)
 	tr.usmMonitor = newUSMMonitor(cfg, tr.ebpfTracer)
 
+	if cfg.EnableProcessEventMonitoring || cfg.EnableUSMEventStream {
```
Comment (hmahmood) [inline pkg/network/tracer/tracer.go:185]: I don't think you need this change as the `events` package only gets used by the network consumer.

Comment (guyarb) [reply pkg/network/tracer/tracer.go:185]: So I need to create a different marker to say "usm is enabled and running" to register my consumer

File: pkg/network/usm/state/state.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	Stopped    MonitorState = "Stopped"
+)
+
+var (
+	State = Disabled
+)
+
+// Set sets the current state of the USM monitor.
+func Set(state MonitorState) {
+	State = state
```
Comment (hmahmood) [inline pkg/network/usm/state/state.go:25]: May want to use atomics here.

**Review by hmahmood — APPROVED** [review]

**Review by paulcacheux — APPROVED** [review]

**Review by vitkyrka — APPROVED** [review]

**Pull request merged**